lib/spack/spack/package.py

```diff
@@ -1828,6 +1836,13 @@ def build_process():
             # check the filesystem for it.
             self.stage.created = False
 
+            # drop into the build environment of the package?
+            if self.drop_in_shell is not None:
```

Comment:
Can you refactor this into the command instead of `package.py`. It seems to me to be part of the interface, not part of the function of building a package.

Reply:
Sounds good. I think the problem was that the package install caused a command exit when run with `until` as soon as the phase was reached... will check.